src/bbs2gh/Factories/AwsApiFactory.cs

```diff
@@ -13,14 +13,8 @@ public AwsApiFactory(EnvironmentVariableProvider environmentVariableProvider)
 
     public virtual AwsApi Create(string awsRegion = null, string awsAccessKeyId = null, string awsSecretAccessKey = null, string awsSessionToken = null)
     {
-#pragma warning disable CS0618
-        awsAccessKeyId ??= _environmentVariableProvider.AwsAccessKeyId(false)
-                           ?? _environmentVariableProvider.AwsAccessKey(false)
-                           ?? _environmentVariableProvider.AwsAccessKeyId();
-        awsSecretAccessKey ??= _environmentVariableProvider.AwsSecretAccessKey(false)
-                               ?? _environmentVariableProvider.AwsSecretKey(false)
-                               ?? _environmentVariableProvider.AwsSecretAccessKey();
-#pragma warning restore CS0618
+        awsAccessKeyId ??= _environmentVariableProvider.AwsAccessKeyId(false);
+        awsSecretAccessKey ??= _environmentVariableProvider.AwsSecretAccessKey(false);
```

Comment:
```suggestion
        awsAccessKeyId ??= _environmentVariableProvider.AwsAccessKeyId();
        awsSecretAccessKey ??= _environmentVariableProvider.AwsSecretAccessKey();
```

The true/false determines whether it throws an exception if  the env var is not set. The previous behavior was:
- Check if the new env var is set but don't throw if it isn't
- Check if the old env var is set but don't throw if it isn't
- Check if the new env var is set (again), but this time throw if it isn't

Reply:
Thanks for flagging! Would this mean that we don't need the validation  code in the `MigrateRepoCommandHandler`? I'm a bit confused about how the two fit together.

Reply:
It looks like this code will run before the validation in the command handler, since the `AwsApi` is passed in when the handler is instantiated.

Reply:
I did a manual test so I'm surprised this is wrong - I guess either my test was totally wrong, or there's a nuanced behaviour difference.

Reply:
This code does run before the handler code, but I think the validation is still needed in the handler for some scenarios.

Reply:
if the secrets are provided via CLI args instead of env vars, the validation in the factory doesn't kick in at all, and validation happens in the handler (and/or args class).

Reply:
Makes sense - I can see how that works in this class. 

I'm happy to make this change, but do you have any idea what the scenario is where we need the validation in the handler?

I'm wondering if we need to keep it at all or if we can simplify.

Reply:
Which specific bit of validation do you think is unnecessary?

Reply:
I'm specifically thinking of this:

https://github.com/github/gh-gei/blob/main/src/bbs2gh/Commands/MigrateRepo/MigrateRepoCommandHandler.cs#L318-L344

I don't think this will be necessary, because `AwsApi` will error if the access key ID and secret access key aren't configured through either the CLI arguments or an environment variable, and this method is called by `ValidateOptions` which is called by `Handle` - and that happens after the handler is instantiated with its own `AwsApi` instance.

Reply:
Yup I think you're right, those validations can never be hit.

Reply:
I think I'll choose to tidy that up in a separate PR, just to avoid polluting this changeset.